02.calendar/cal.rb

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'date'
+require 'optparse'
+
+params = ARGV.getopts('', 'y:', 'm:')
+
+today = Date.today
+
+def print_calendar(year, month)
+  start_date = Date.new(year, month, 1)
+  end_date = Date.new(year, month, -1)
+  puts "#{month}月 #{year}".center(20)
+  puts '日 月 火 水 木 金 土'
+  print '   ' * start_date.wday
+
+  (start_date..end_date).each do |date|
+    print "#{date.day.to_s.rjust(2)} "
+    print "\n" if date.saturday?
+  end
+  print "\n"
+end
+
+year = params['y'] ? params['y'].to_i : today.year
```

Comment:
```suggestion
params = ARGV.getopts('', 'y:', 'm:')
today = Date.today
year = params['y'] ? params['y'].to_i : today.year
```

変数の寿命を短くするならparamsとtodayもここにないとダメですね

あと高等テクニックですがこんな書き方もできます。

```ruby
year = params['y']&.to_i || today.year
```

Reply:
なるほど。そんな書き方もあるんですね。
勉強になります！修正してからマージします！